lib/date_time/Kconfig

```diff
@@ -9,6 +9,13 @@ menuconfig DATE_TIME
 
 if DATE_TIME
 
+config DATE_TIME_AUTO
+	bool "Automatically update time"
```

Comment:
Currently, this also affects the %XTIME notification usage. I'm wondering whether the library could always use the %XTIME notification, because that doesn't cost anything. This option (the name should probably be changed) could be used only for synchronizing the time automatically during startup. This could result in use of NTP, if network time is not available.

Reply:
True, we could use XTIME without this AUTO update feature (whatever the flag will be). XTIME feature should be flagged with `CONFIG_DATE_TIME_MODEM` rather than `CONFIG_DATE_TIME_AUTO`.

Reply:
My current proposal for `CONFIG_DATE_TIME_AUTO` is `CONFIG_DATE_TIME_SYSTEM_UPDATE` and maybe do another one for modem time updates `CONFIG_DATE_TIME_MODEM_UPDATE`.
"store" is one alternative for "update".

Reply:
Should `CONFIG_DATE_TIME_SYSTEM_UPDATE` actually only control whether we trigger time update at startup (in which case it should be renamed), or should it also control whether we ever update Zephyr and modem time? If it controls whether Zephyr time is updated, I think we cannot remove `time_aux` as we wouldn't have storage for time:
https://github.com/nrfconnect/sdk-nrf/pull/6125#discussion_r754825802

Reply:
Based on discussions, I've changed this to `CONFIG_DATE_TIME_AUTO_UPDATE` and it only applies as trigger for initial time retrieval when LTE connection is first created.

Reply:
@tokangas, Please check if this can be closed.